signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/azure/AzureKeyVaultSigner.java

```diff
@@ -58,13 +66,14 @@ public class AzureKeyVaultSigner implements Signer {
         useDeprecatedSignatureAlgorithm
             ? SignatureAlgorithm.fromString("ECDSA256")
             : SignatureAlgorithm.ES256K;
-  }
-
-  @Override
-  public Signature sign(byte[] data) {
-    final AzureKeyVault vault;
+    final AzureConnectionFactory connFactory = new AzureConnectionFactory();
+    final AzureConnectionParameters connectionParameters =
+        AzureConnectionParameters.newBuilder()
+            .withServerHost(constructAzureKeyVaultUrl(config.getKeyVaultName()))
+            .build();
+    this.azureConnection = connFactory.getOrCreateConnection(connectionParameters);
```

Comment:
Do we end up with an AzureConnection object per Signer for the entire lifetime of the app? What happens when there's 10,000 signers?

Reply:
I was thinking a connection pool might be appropriate but not sure what perf tests you've already done, so might be unnecessary

Reply:
Changed it to use the connection pool that we're passing to that class now